library/core/src/pin.rs

```diff
@@ -414,11 +414,11 @@ pub struct Pin<P> {
     // Long-term, `unsafe` fields or macro hygiene are expected to offer more robust alternatives.
     #[unstable(feature = "unsafe_pin_internals", issue = "none")]
     #[doc(hidden)]
-    pub pointer: P,
+    pub __pointer: P,
```

Comment:
can you leave a comment mentioning the deref impact of the field?

Reply:
I've added a comment.